provisioner/data.js

```diff
@@ -1,30 +1,704 @@
+     * be used.  In each instance type, there are three keys:
+     *   - capacity: this is the number of tasks this instance
+     *               can run concurrently.
+     *   - utility: this is an arbitrary number which we multiply
+     *              by spot price and compare to other instance
+     *              types to figure out which machine to bid on
+     *   - overwrites: this object overwrites keys in the general
+     *                 launch specification
+     */
+    types: base.Entity.types.JSON,
+    /* This is a JSON object which contains the regions which this
+     * workerType is allowed to run in as well as the region-specific
+     * information.  It is in the shape:
+     * {
+     *   'us-west-1': {
+     *     'overwrites': {
+     *       'ImageId': 'ami-aaaaaaa'
+     *      }
+     *   }
+     * } */
+    regions: base.Entity.types.JSON,
+  },
+  context: ['ec2', 'keyPrefix', 'pubKey', 'influx'],
 });
 
-// Export WorkerType
-exports.WorkerType = WorkerType;
\ No newline at end of file
+/**
+ * Create a workerType in the table.  The properties
+ * should not have a workerType key since that will be
+ * specified in the workerType argument
+ */
+WorkerType.create = function(workerType, properties) {
+  assert(workerType);
+  assert(properties);
+  assert(!properties.workerType);
+  properties.workerType = workerType;
+  return base.Entity.create.call(this, properties);
+};
+
+
+/**
+ * Return a list of all known workerTypes
+ */
+WorkerType.loadAll = function() {
+  var workers = [];
+
+  var p = base.Entity.scan.call(this, {}, {
+    handler: function (item) {
+      workers.push(item);
+    }
+  });
+
+  p = p.then(function() {
+    return workers;
+  });
+
+  return p;
+};
+
+/**
+ * Load the names of all known workerTypes
+ */
+WorkerType.listWorkerTypes = function() {
+  var names = [];
+
+  var p = base.Entity.scan.call(this, {}, {
+    handler: function (item) {
+      names.push(item.workerType);
+    }
+  });
+
+  p = p.then(function() {
+    return names;
+  });
+
+  return p;
+};
+
+
+/**
+ * Turn off every single EC2 instance and cancel all spot
+ * requests which were created by this Provisioner
+ */
+WorkerType.killEverything = function (debug) {
+  assert(debug);
+
+  var p = WorkerType.loadAll.call(this);
+
+  p = p.then(function (workerTypes) {
+    return Promise.all(workerTypes.map(function(workerType) {
+      return workerType.killAll(debug);
+    }));
+  });
+
+  return p;
+};
+
+
+/**
+ * Load a single workerType by name
+ */
+WorkerType.load = function(workerType) {
+  assert(workerType);
+  return base.Entity.load.call(this, {
+    workerType: workerType
+  });
+};
+
+
+/**
+ * Return an Object for JSON encoding which represents
+ * the data associated with this WorkerType.  This is a
+ * method intended for use in displaying the data associated
+ * with a given workerType
+ */
+WorkerType.prototype.json = function() {
+  return lodash.clone(this.__properties);
```

Comment:
`base.Entity` does support non-json types like `Date` objects and buffers.

Besides is very rare that what we store in table storage matches what we return... Ususally we do some modifications because we store in a slightly different format, or we don't store redundant information...In almost all cases we have internal state that we don't return the users.

I suspect it's just a matter of time before you'll have that here too.